pkg/subctl/cmd/validate_deployment.go

```diff
@@ -140,3 +145,33 @@ func CheckDaemonset(k8sClient kubernetes.Interface, namespace, daemonSetName str
 
 	return true
 }
+
+func checkPodsStatus(k8sClient kubernetes.Interface, operatorNamespace string) bool {
+	pods, err := k8sClient.CoreV1().Pods(operatorNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		message := fmt.Sprintf("Error obtaining Pods list: %v", err)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return false
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != v1.PodRunning {
+			message := fmt.Sprintf("Pod %q is not running. (current state is %v)", pod.Name, pod.Status.Phase)
+			status.QueueFailureMessage(message)
+			status.End(cli.Failure)
+			return false
+		}
+
+		for _, c := range pod.Status.ContainerStatuses {
+			if c.RestartCount >= 5 {
+				message := fmt.Sprintf("Pod %q has restarted %d times", pod.Name, c.RestartCount)
+				status.QueueWarningMessage(message)
+				status.End(cli.Failure)
```

Comment:
Since the pod is currently in Running State and restartCount is just a Warning, I feel we can return `status.End(cli.Success)` instead of Failure.

Reply:
@sridhargaddam Do you think we should still return false in this case?
maybe we should skip the return line from this loop so it will check other pods as well?

Reply:
Yes, agree with you.